mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -0,0 +1,142 @@
+          TypeName propertyType = TypeName.get(returnTypeMirror);
+          if (TypeName.VOID.equals(propertyType)) {
+            continue;
+          }
+          String propertyName = Introspector.decapitalize(methodName.substring(3));
+          DefaultPropertyDefinition.Builder builder = propertyBuilders.get(propertyName);
+          if (builder == null) {
+            builder =
+                new DefaultPropertyDefinition.Builder(
+                    propertyName, propertyType, getEntityElement(returnTypeMirror));
+            propertyBuilders.put(propertyName, builder);
+          } else if (!builder.getType().equals(propertyType)) {
+            context
+                .getMessager()
+                .warn(
+                    method,
+                    "Ignoring method %s %s() because there is a setter "
+                        + "with the same name but a different type: %s(%s)",
+                    propertyType,
+                    methodName,
+                    builder.getSetterName(),
+                    builder.getType());
+            continue;
+          }
+          builder.withGetterName(methodName);
+        } else if (methodName.startsWith("set") && method.getParameters().size() == 1) {
+          String propertyName = Introspector.decapitalize(methodName.substring(3));
+          VariableElement parameter = method.getParameters().get(0);
+          TypeMirror typeMirror = parameter.asType();
+          TypeName propertyType = TypeName.get(typeMirror);
+          DefaultPropertyDefinition.Builder builder = propertyBuilders.get(propertyName);
+          if (builder == null) {
+            builder =
+                new DefaultPropertyDefinition.Builder(
+                    propertyName, propertyType, getEntityElement(typeMirror));
+            propertyBuilders.put(propertyName, builder);
+          } else if (!builder.getType().equals(propertyType)) {
+            context
+                .getMessager()
+                .warn(
+                    method,
+                    "Ignoring method %s(%s) because there is a getter "
+                        + "with the same name but a different type: %s %s()",
+                    methodName,
+                    propertyType,
+                    builder.getType(),
+                    builder.getGetterName());
+            continue;
+          }
+          builder.withSetterName(methodName);
+        }
+      }
+    }
+
+    ImmutableList.Builder<PropertyDefinition> definitions = ImmutableList.builder();
+    for (DefaultPropertyDefinition.Builder builder : propertyBuilders.values()) {
+      if (builder.getGetterName() != null && builder.getSetterName() != null) {
+        definitions.add(builder.build());
+      }
+    }
+
+    String entityName = Introspector.decapitalize(classElement.getSimpleName().toString());
+    return new DefaultEntityDefinition(entityName, definitions.build());
+  }
+
+  private TypeElement getEntityElement(TypeMirror typeMirror) {
```

Comment:
Maybe we could wrap raw `TypeElement` into class `EntityElement` to have more strong types?
We could wrap all specific `TypeElement`s this way to make our code more explicit about types that is able to handle

Reply:
I'm not sure if we can extract anything higher-level than `TypeMirror`/`Element`; I feel we would end up duplicating most of the API: at the end of the day we still need to handle all the weird cases, like missing types, etc.

It's still a bit early anyway. Let's wait for a few more tickets and maybe something will emerge.